rules/terraformrules/terraform_unused_declaration.go

```diff
@@ -105,7 +105,7 @@ func (r *TerraformUnusedDeclarationsRule) declarations(module *configs.Module) *
 func (r *TerraformUnusedDeclarationsRule) checkForRefsInExpr(expr hcl.Expression, decl *declarations) error {
 	refs, diags := lang.ReferencesInExpr(expr)
 	if diags.HasErrors() {
-		return diags.Err()
+		return nil
```

Comment:
Logging errors instead of just ignoring them may help in investigating when problems occur.

Reply:
Adding a log message now